pip wheels for mac broken

Looks like our x86 wheels for OSX are broken and have been for a bit:
```
../../../hostedtoolcache/Python/3.9.21/x64/lib/python3.9/site-packages/msprime/__init__.py:28: in <module>
28
    from msprime._msprime import (
29
E   ImportError: dlopen(/Users/runner/hostedtoolcache/Python/3.9.21/x64/lib/python3.9/site-packages/msprime/_msprime.cpython-39-darwin.so, 0x0002): symbol not found in flat namespace '_gsl_rng_default'
30
```

Would you mind having a look at this please @benjeffery and seeing if there's a reasonable fix?

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -12,9 +12,11 @@ on:
 
 jobs:
   OSX:
-    runs-on: macos-latest
+    name: Build ${{ matrix.os }} - ${{ matrix.python }}
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [macos-13, macos-15]
         python: [3.9, "3.10", 3.11, 3.12]
     steps:
       - name: Checkout
@@ -29,13 +31,21 @@ jobs:
         run: |
           brew install gsl
           pip install --upgrade pip build
-      - name: Build Wheel
+      - name: Build Wheel (arm64)
+        if: matrix.os == 'macos-15'
         run: |
           export PLAT="arm64"
           export _PYTHON_HOST_PLATFORM="macosx-11.0-arm64"
-          export OLDARCHFLAGS=$ARCHFLAGS
-          export ARCHFLAGS+="-arch arm64"
+          export CFLAGS+=" -arch arm64"
+          export CXXFLAGS+=" -arch arm64"
+          export ARCHFLAGS=$OLDARCHFLAGS
+          export ARCHFLAGS+=" -arch arm64"
+          export CPPFLAGS+=" -arch arm64"
+          export LDFLAGS+=" -arch arm64"
           python -m build --wheel
+      - name: Build Wheel (x86_64)
+        if: matrix.os == 'macos-13'
+        run: |
           export PLAT="x86_64"
           export _PYTHON_HOST_PLATFORM="macosx-11.0-x86_64"
           export CFLAGS+=" -arch x86_64"
@@ -48,12 +58,11 @@ jobs:
       - name: Delocate to bundle dynamic libs
         run: |
           pip install delocate
-          delocate-wheel -v dist/msprime-*arm64.whl
-          delocate-wheel -v dist/msprime-*x86_64.whl
+          delocate-wheel -v dist/*.whl
       - name: Upload Wheels
         uses: actions/upload-artifact@v4.6.1
         with:
-          name: osx-wheel-${{ matrix.python }}
+          name: osx-wheel-${{ matrix.os }}-${{ matrix.python }}
           path: dist
 
   manylinux:
@@ -89,16 +98,18 @@ jobs:
           path: dist/wheelhouse
 
   OSX-test:
+    name: Test ${{ matrix.os }} - ${{ matrix.python }}
     needs: ['OSX']
-    runs-on: macos-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [macos-13, macos-15]
         python: [3.9, "3.10", 3.11, 3.12]
     steps:
       - name: Download wheels
         uses: actions/download-artifact@v4.2.0
         with:
-          name: osx-wheel-${{ matrix.python }}
+          name: osx-wheel-${{ matrix.os }}-${{ matrix.python }} 
       - name: Set up Python ${{ matrix.python }}
         uses: actions/setup-python@v5.4.0
         with:

